components/org.wso2.carbon.identity.verification.onfido.connector/src/main/java/org/wso2/carbon/identity/verification/onfido/connector/web/OnfidoAPIClient.java

```diff
@@ -62,12 +65,37 @@ public static JSONObject createApplicant(Map<String, String> idVConfigPropertyMa
         if (response.getStatusLine().getStatusCode() == HttpStatus.SC_CREATED) {
             return getJsonObject(response);
         } else {
-            throw new IdentityVerificationServerException(ERROR_CREATING_ONFIDO_APPLICANT.getCode(),
+            throw new OnfidoServerException(ERROR_CREATING_ONFIDO_APPLICANT.getCode(),
                     String.format(ERROR_CREATING_ONFIDO_APPLICANT.getMessage(),
                             response.getStatusLine().getStatusCode()));
         }
     }
 
+    /**
+     * This method is used to create a workflow run in Onfido.
+     *
+     * @param idVConfigPropertyMap The map of the configuration properties.
+     * @param workflowRunRequestBody The request body for creating the workflow run in JSON format.
+     * @return The created workflow run details as a JSONObject.
+     * @throws OnfidoServerException If an error occurs while creating the workflow run on the Onfido server.
+     */
+    public static JSONObject createWorkflowRun(Map<String, String> idVConfigPropertyMap,
+                                               JSONObject workflowRunRequestBody)
+            throws OnfidoServerException {
+
+        String apiToken = idVConfigPropertyMap.get(TOKEN);
```

Comment:
Shall we check if the `idVConfigPropertyMap` is null since these are Public methods?

Reply:
The methods defined in the `OnfidoAPIClient` class are only used within the `OnfidoIdentityVerifier` class. The `idVConfigPropertyMap` is already checked for null via this [method](https://github.com/wso2-extensions/identity-verification-onfido/blob/6f73cff0b813bfcfffbd6cfbe9d730df1f5fe3ed/components/org.wso2.carbon.identity.verification.onfido.connector/src/main/java/org/wso2/carbon/identity/verification/onfido/connector/OnfidoIdentityVerifier.java#L495) before being passed into these methods. While these methods are public, they are not intended to be accessed by other external components, so adding additional null checks here would be redundant in my point of view.